test/Integration/IntegrationTestBase.cs

```diff
@@ -188,26 +188,33 @@ protected void StartFunctionHost(string functionName, SupportedLanguages languag
             {
                 StartInfo = startInfo
             };
+
+            // Register all handlers before starting the functions host process.
+            var taskCompletionSource = new TaskCompletionSource<bool>();
             this.FunctionHost.OutputDataReceived += this.TestOutputHandler;
+            this.FunctionHost.OutputDataReceived += SignalStartupHandler;
+            this.FunctionHost.OutputDataReceived += customOutputHandler;
+
             this.FunctionHost.ErrorDataReceived += this.TestOutputHandler;
 
             this.FunctionHost.Start();
             this.FunctionHost.BeginOutputReadLine();
             this.FunctionHost.BeginErrorReadLine();
 
-            var taskCompletionSource = new TaskCompletionSource<bool>();
-            this.FunctionHost.OutputDataReceived += (object sender, DataReceivedEventArgs e) =>
+            this.TestOutput.WriteLine($"Waiting for Azure Function host to start...");
+            taskCompletionSource.Task.Wait(60000);
+            this.TestOutput.WriteLine($"Azure Function host started!");
+            this.FunctionHost.OutputDataReceived -= SignalStartupHandler;
+
+            void SignalStartupHandler(object sender, DataReceivedEventArgs e)
             {
                 // This string is printed after the function host is started up - use this to ensure that we wait long enough
                 // since sometimes the host can take a little while to fully start up
-                if (e != null && !string.IsNullOrEmpty(e.Data) && e.Data.Contains($"http://localhost:{this.Port}/api"))
+                if (e.Data?.Contains(" Host initialized ") == true)
```

Comment:
Updated the log line that gets waited for to signal that the Functions Host has initialized. The old line works only for functions with HTTP triggers. As more tests are getting added, waiting for 60 seconds for each test was making the overall running time for tests slower. I have verified that none of the tests fail or wait until timeout due to this change.